whd/webhook.py

```diff
@@ -33,7 +34,7 @@ def __init__(
         self.whd_cfg = whd_cfg
         self.dispatcher = GithubWebhookDispatcher(cfg_set=cfg_set, whd_cfg=whd_cfg)
 
-    def on_post(self, req, resp):
+    def on_post_handling(self, req, resp):
```

Comment:
firstly, please do not name the method like that - as per falcon's convention, this method name suggests that this be hooked to a POST with "suffix" `_handling`. marking the method as private would be the better choice (e.g. _on_post)

also, this is not the suggested semantics, as this will not return early and thus avoid timeouts.

please do the following:

- identify the long-running part of the coding
- define an inline function handling this part
- start a thread, passing said inline function as target
- after starting the thread, return http-200